src/main/java/com/auth0/net/Request.java

```diff
@@ -3,6 +3,8 @@
 import com.auth0.exception.APIException;
 import com.auth0.exception.Auth0Exception;
 
+import java.util.concurrent.CompletableFuture;
```

Comment:
This class is not on JDK 7 (min for this library). You might want to use some other `Future`.

Reply:
Java `Future` is not an equivalent to `CompletableFuture`, so no.

Reply:
@guizmaii Then no, I won't merge this PR unless a compatible class is used. We still support JDK 7 like I said, releasing this would be a breaking change for most of our customers. If in the future we plan a new major we would definitely add something like this.